tree/ntuple/v7/src/RField.cxx

```diff
@@ -2412,17 +2415,14 @@ void ROOT::Experimental::RField<std::vector<bool>>::GenerateColumnsImpl(const RN
 std::vector<ROOT::Experimental::Detail::RFieldBase::RValue>
 ROOT::Experimental::RField<std::vector<bool>>::SplitValue(const RValue &value) const
 {
-   const static bool trueValue = true;
-   const static bool falseValue = false;
-
    auto typedValue = value.Get<std::vector<bool>>();
    auto count = typedValue->size();
    std::vector<RValue> result;
    for (unsigned i = 0; i < count; ++i) {
       if ((*typedValue)[i])
-         result.emplace_back(fSubFields[0]->BindValue(const_cast<bool *>(&trueValue)));
+         result.emplace_back(fSubFields[0]->BindValue(std::shared_ptr<bool>(new bool(true))));
       else
-         result.emplace_back(fSubFields[0]->BindValue(const_cast<bool *>(&falseValue)));
+         result.emplace_back(fSubFields[0]->BindValue(std::shared_ptr<bool>(new bool(false))));
```

Comment:
Not sure if this is on the critical path? Maybe we'd want to optimize this to avoid `new` on every call... (not sure if a `std::shared_ptr` with an empty deleter would work)